### DISCUSSION TITLE
Don't show current project in recent projects.

### DISCUSSION THREAD

**Discussion by VladasZ** [zed-industries/zed#12497] [Show and tell]
Recent projects is very cool feature. But by default the first project in the recent list is current project which is a bit inconvenient in my opinion. I need to press another button to switch to last opened project. Why would anyone want to open the project which is already in front of you? This action does nothing.

I think it is better to hide current project from this list.

I implemented a very simple skip here:
https://github.com/VladasZ/zed/commit/639164d8771e286a60ec2265aa300390d9d48b5b

And now recent project list doesn't show current project.

I want to discuss if this is a good idea in general. 
Or maybe it is better to put current project at the end of the list but still display it?
Or this should be a new option?

**Comment by SomeoneToIgnore** [comment]
The idea looks nice, but I think we have to do more and remove all related rendering logic in https://github.com/zed-industries/zed/blob/230b10d3c4ce2c90383752ddbcf02c49728e267a/crates/recent_projects/src/recent_projects.rs#L397

Also `.skip(1)` looks dangerous, I would use a filter with the `is_current_workspace` function instead.

**Reply by VladasZ** [reply]
Yeah I agree. Here is the better implementation: https://github.com/VladasZ/zed/commit/168cebf7715e4fb1cc110f0aa7b7c624ecbc95d3

**Reply by SomeoneToIgnore** [reply]
Style-wise, I think it's more readable when split into `.filter().map()` chain, but otherwise looks like a PR to me.
As a bonus change since we're touching that line, one could also replace `&self.workspaces[hit.candidate_id];` with `.get()?` to be 100% sure that Zed won't panic here ever.

**Reply by VladasZ** [reply]
Okay I'll do that and create a PR.

**Reply by VladasZ** [reply]
@SomeoneToIgnore : https://github.com/zed-industries/zed/pull/12502
